woocommerce/woocommerce-gateway-stripe
# Add account summary API endpoint

Fixes #2121

## Changes proposed in this Pull Request:

For Stripe Terminal support, the mobile apps need to be able to get some details of the Stripe account. The Stripe extension already has an API endpoint (`wc_stripe/account`) to get full account details, but the mobile apps need the data returned in the same format that WCPay uses in its own account endpoint (WCPay does some additional processing to determine whether account requirements are met).

This PR adds another endpoint (`wc_stripe/account/summary`) to return account details in the same format as WCPay.

Additional Context: pdfdoF-er-p2

## Testing instructions

For making API requests, I found it helpful to install the [Basic Auth](https://github.com/WP-API/Basic-Auth) plugin to my test site so that I could make requests of the form `curl --user username:password 'http://YOUR_SITE/wp-json/wc/v3/wc_stripe/account/summary'` to test the endpoint.

1. Run `curl --user username:password 'http://YOUR_SITE/wp-json/wc/v3/wc_stripe/account/summary'`
2. Verify that the response you get matches the format of the response under the `data` key in the "Load Account" section of pdfdoF-er-p2.
3. To test making a change to the account details, go to your Stripe settings (https://dashboard.stripe.com/settings/account) and change your statement descriptor.
4. Delete the `_transient_wcstripe_account_data_test` transient from the `wp_options` table to force the API to load the data from Stripe on the next request.
5. Run `curl --user username:password 'http://YOUR_SITE/wp-json/wc/v3/wc_stripe/account/summary'` again, verify that you see your new statement descriptor in the response.

---


**Post merge**
